EDDiscovery/TravelHistoryControl.cs

```diff
@@ -665,10 +658,11 @@ private void buttonUpdate_Click(object sender, EventArgs e)
                 distance.Store();
                 SQLiteDBClass.AddDistanceToCache(distance);
 
-                dataGridViewTravel.Rows[lastRowIndex].Cells[2].Value = textBoxDistance.Text.Trim();
+                if (dataGridViewTravel.SelectedCells.Count > 0)          // if we have selected (we should!)
+                    dataGridViewTravel.Rows[dataGridViewTravel.SelectedCells[0].OwningRow.Index].Cells[2].Value = textBoxDistance.Text.Trim();
```

Comment:
Trivial: Personally I'd extract the `dataGridViewTravel.SelectedCells[0].OwningRow.Index` bit into a variable or method for better readability